protostar/cheatable_starknet/cheatcodes/cairo_cheatcode.py

```diff
@@ -1,12 +1,65 @@
-from abc import ABC
-from typing import TYPE_CHECKING
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Union
+
+from starknet_py.cairo.felt import encode_shortstring
 
 from protostar.cairo import HintLocal
+from protostar.cheatable_starknet.cheatable_exception import CheatableException
 
 if TYPE_CHECKING:
     from protostar.cheatable_starknet.cheaters import CairoCheaters
 
+Felt = int
+
+
+@dataclass
+class CheatcodeException(Exception):
+    code: Felt
+    ex: Exception
+
+
+@dataclass
+class CairoCheatcodeInvalidExecution:
+    ok = None
+    err: CheatcodeException
+
+    def unwrap(self) -> Any:
+        raise self.err
+
+
+@dataclass
+class CairoCheatcodeValidExecution:
+    ok: Any
+    err = None
+
+    def unwrap(self) -> Any:
+        return self.ok
```

Comment:
Shouldn't `unwrap` be only callable on a dataclass with valid result? The code will result in an error if we unwrap and then pass the error as a "result" to later calls. We should probably "panic" when unwrapping a result which has an error underneath, so that the code will crash on unwrap instead of a typeerror when passing error to further functions

Reply:
> Shouldn't unwrap be only callable on a dataclass with valid result? The code will result in an error if we unwrap and then pass the error as a "result" to later calls.

`CairoCheatcodeInvalidExecution::unwrap` raises (not returns) an error, which should result in panic.

Reply:
Removed `unwrap`.